components/style/properties/properties.mako.rs

```diff
@@ -969,6 +969,29 @@ impl<T> DeclaredValueOwned<T> {
     }
 }
 
+/// A cache for property declarations parsed from a shorthand.
+pub struct ParsedShorthandCache<'a> {
```

Comment:
Is there any chance to move this to the `custom_properties` module? Probably `UnparsedValue` should be moved to, just leaving the code that needs mako here.

Otherwise it's harder to review and touch (I was just playing with it locally).

WDYT?

Reply:
I think it makes sense. The problem is, in that case, we would need to make all its field public, otherwise the method in this mod wouldn't be able to access it. I guess that's not a big deal, though.